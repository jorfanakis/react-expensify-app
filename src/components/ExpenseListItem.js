import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

export const ExpenseListItem = ({amount, createdAt, description, id, note}) => (
  <div>
    <ul>
      <Link to={`/edit/${id}`}>
        <li>{description}</li>
      </Link>
      <p>
        {numeral(amount / 100).format('$0,0.00')}
        -
        {moment(createdAt).format('MMMM Do, YYYY')}
        -
        {note}
      </p>
    </ul>
 </div>
);

export default ExpenseListItem;
