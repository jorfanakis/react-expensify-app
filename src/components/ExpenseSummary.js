import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import expenseSummary from '../selectors/expenses-total';

export const ExpenseSummary = ({ expenseCount, expenseTotal }) => {
  const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
  const formattedTotal = numeral(expenseTotal / 100).format('$0,0.00');
  return (
    <div>
    {
      <h1>Viewing {expenseCount} {expenseWord} totalling {formattedTotal}</h1>
    }
    </div>
  );
};

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);
  return {
    expenseCount: visibleExpenses.length,
    expenseTotal: expenseSummary(visibleExpenses)
  };
};

const ConnectedExpenseSummary = connect(mapStateToProps)(ExpenseSummary);

export default ConnectedExpenseSummary;