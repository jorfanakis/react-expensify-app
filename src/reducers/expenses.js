
const expensesReducerDefaultState = [];
const expensesReducer = (state = expensesReducerDefaultState,  action) => {
  switch (action.type) {
    case 'ADD_EXPENSE':
      //return state.concat(action.expense);
      return [
        ...state,
        action.expense
      ];

    case 'REMOVE_EXPENSE':
      //return state.filter((e) => { return e.id !== action.id; });
      return state.filter(({ id }) => id !== action.id);

    case 'EDIT_EXPENSE':
      return state.map((e) =>{
        if (e.id === action.id) {
          return {
            ...e,
            ...action.updates
          };
        } else {
          return e;
        }
      });

  case 'SET_EXPENSES':
    return action.expenses;

    default:
      return state;
  }

};

export default expensesReducer;
