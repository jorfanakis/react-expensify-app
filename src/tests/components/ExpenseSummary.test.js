import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseSummary } from '../../components/ExpenseSummary';
import expenses from '../fixtures/expenses';

test('should render ExpenseSummary with no expenses', () => {
  const wrapper = shallow(<ExpenseSummary expenseCount={0} expenseTotal={234} />);
  expect(wrapper).toMatchSnapshot()
});

test('should render ExpenseSummary with one expense', () => {
  const wrapper = shallow(<ExpenseSummary expenseCount={1} expenseTotal={expenses[0].amount} />);
  expect(wrapper).toMatchSnapshot()
});

test('should render ExpenseSummary with expenses', () => {
  const wrapper = shallow(<ExpenseSummary expenseCount={23} expenseTotal={1230947327} />);
  expect(wrapper).toMatchSnapshot()
});
