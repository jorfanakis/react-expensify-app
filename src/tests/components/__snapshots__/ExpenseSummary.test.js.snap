// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary with expenses 1`] = `
<div>
  <h1>
    Viewing 
    23
     
    expenses
     totalling 
    $12,309,473.27
  </h1>
</div>
`;

exports[`should render ExpenseSummary with no expenses 1`] = `
<div>
  <h1>
    Viewing 
    0
     
    expenses
     totalling 
    $2.34
  </h1>
</div>
`;

exports[`should render ExpenseSummary with one expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $1.95
  </h1>
</div>
`;
